CRM/Contribute/BAO/Product.php

```diff
@@ -88,31 +82,28 @@ public static function setIsActive($id, $is_active) {
   /**
    * Add a premium product to the database, and return it.
    *
-   * @param array $params
-   *   Reference array contains the values submitted by the form.
-   * @param array $ids
-   *   Reference array contains the id.
+   * @param array $productParams
+   *   contains the values submitted by the form.
    *
    * @return CRM_Contribute_DAO_Product
    */
-  public static function add(&$params, &$ids) {
-    $params = array_merge(array(
-      'id' => CRM_Utils_Array::value('premium', $ids),
+  public static function add($productParams) {
+    $productParams = array_merge(array(
```

Comment:
$params is really what we normally call this array on create methods - I think we should keep it. Also, once we get fully stdised we should add a hook - I think we should do an interim step & deprecate rather than remove $ids - https://github.com/civicrm/civicrm-core/pull/12451/files

We don't support extensions calliing these things directly but we do try to deprecate then remove when possible to reduce possible pain